src/librustc_save_analysis/lib.rs

```diff
@@ -622,9 +622,17 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
             Node::Visibility(&Spanned {
                 node: hir::VisibilityKind::Restricted { ref path, .. }, .. }) => path.def,
 
-            Node::PathSegment(seg) => match seg.def {
-                Some(def) => def,
-                None => HirDef::Err,
+            Node::PathSegment(seg) => {
+                let def = match seg.def {
```

Comment:
Is this equivalent to:
```rust
match seg.def {
    Some(def) if def != HirDef::Err => def,
    _ => self.get_path_def(self.tcx.hir().get_parent_node(id)),
}
```
And if so, could we keep the concise form? I believe it reads better.

Reply:
Sure, I can write it that way.